cuebot/src/main/java/com/imageworks/spcue/dispatcher/FrameCompleteHandler.java

```diff
@@ -175,6 +176,11 @@ public void run() {
                 }
             }
             else {
+                if (dispatchSupport.terminateFrame(frame, FrameState.WAITING, report.getExitStatus(),
+                    report.getFrame().getMaxRss())) {
+
+                    logger.info("stopped terminating frame: " + frame.state + " -> " + newFrameState);
+                }
```

Comment:
The above cleans up the `TERMINATING` state and sets it to `WAITING`.  This path is executed for all frames that weren't `RUNNING` (like killed frames).  After this point, everything will behave as it did prior to modification.